v2/car.go

```diff
@@ -94,21 +106,38 @@ func (h Header) WriteTo(w io.Writer) (n int64, err error) {
 		return
 	}
 	n += wn
-	if err = writeUint64To(w, h.CarV1Offset); err != nil {
+	if err = binary.Write(w, binary.LittleEndian, h.CarV1Offset); err != nil {
 		return
 	}
 	n += uint64BytesSize
-	if err = writeUint64To(w, h.CarV1Size); err != nil {
+	if err = binary.Write(w, binary.LittleEndian, h.CarV1Size); err != nil {
 		return
 	}
 	n += uint64BytesSize
-	if err = writeUint64To(w, h.IndexOffset); err != nil {
+	if err = binary.Write(w, binary.LittleEndian, h.IndexOffset); err != nil {
 		return
 	}
 	n += uint64BytesSize
 	return
 }
 
-func writeUint64To(w io.Writer, v uint64) error {
-	return binary.Write(w, binary.LittleEndian, v)
+func (h Header) ReadFrom(r io.Reader) (n int64, err error) {
+	rn, err := h.Characteristics.ReadFrom(r)
+	if err != nil {
+		return
+	}
+	n += rn
+	if err = binary.Read(r, binary.LittleEndian, &h.CarV1Offset); err != nil {
```

Comment:
perf nit, for once you start benchmarking: binary.Read also uses a type switch on the last parameter, which is an interface, so you do quite a bit of branching for every integer you parse. it will likely be faster to use binary.LittleEndian.Uint64 once you read the entire header as a []byte.